src/main/java/gyro/google/compute/RegionInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,193 @@
+                List<ComputeInstanceGroupManagerAutoHealingPolicy> diffableAutoHealingPolicy = getAutoHealingPolicy();
+                List<InstanceGroupManagerAutoHealingPolicy> autoHealingPolicies = null;
+
+                if (diffableAutoHealingPolicy.isEmpty()) {
+                    autoHealingPolicies = Data.nullOf(List.class);
+                } else {
+                    autoHealingPolicies = diffableAutoHealingPolicy
+                        .stream()
+                        .map(ComputeInstanceGroupManagerAutoHealingPolicy::copyTo)
+                        .collect(Collectors.toList());
+                }
+                instanceGroupManager.setAutoHealingPolicies(autoHealingPolicies);
+                shouldPatch = true;
+            } else if (changedFieldName.equals("named-port")) {
+                // TODO: investigate as GCP UI allows updating of named ports.
+            }
+        }
+
+        if (shouldPatch) {
+            patch(instanceGroupManager);
+        }
+    }
+
+    @Override
+    public void doDelete(GyroUI ui, State state) throws Exception {
+        Compute client = createClient(Compute.class);
+        Operation operation = client.regionInstanceGroupManagers()
+            .delete(getProjectId(), getRegion(), getName())
+            .execute();
+        waitForCompletion(client, operation);
+    }
+
+    @Override
+    public void copyFrom(InstanceGroupManager model) {
+        super.copyFrom(model);
+
+        setDistributionPolicy(Optional.ofNullable(model.getDistributionPolicy())
+            .map(e -> {
+                ComputeDistributionPolicy computeDistributionPolicy = newSubresource(
+                    ComputeDistributionPolicy.class);
+                computeDistributionPolicy.copyFrom(e);
+                return computeDistributionPolicy;
+            })
+            .orElse(null));
+        // Do NOT update region with a full url as this should be a name.
+        //        setRegion(model.getRegion());
+    }
+
+    private void setInstanceTemplate() throws Exception {
+        Compute client = createClient(Compute.class);
+        RegionInstanceGroupManagersSetTemplateRequest request = new RegionInstanceGroupManagersSetTemplateRequest();
+        request.setInstanceTemplate(getInstanceTemplate().getSelfLink());
+        Operation operation = client.regionInstanceGroupManagers()
+            .setInstanceTemplate(getProjectId(), getRegion(), getName(), request)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
+    }
+
+    private void patch(InstanceGroupManager instanceGroupManager) throws Exception {
+        Compute client = createClient(Compute.class);
+        Operation operation = client.regionInstanceGroupManagers()
+            .patch(getProjectId(), getRegion(), getName(), instanceGroupManager)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
```

Comment:
Refresh should be called at the end the create/update unless you need to reset a value before.